packages/components/src/heading/types.ts

```diff
@@ -17,10 +17,7 @@ export type HeadingSize =
 	| '5'
 	| '6';
 
-export type HeadingProps = Omit<
-	TextProps,
-	'size' | 'isBlock' | 'color' | 'weight'
-> & {
+export type HeadingProps = Omit< TextProps, 'isBlock' | 'color' | 'weight' > & {
```

Comment:
The "size" definitely impacts the heading rendering, so it's a valid `HeadingProps`. cc @ciampo